csi/server/plugin/opensds/node.go

```diff
@@ -54,17 +55,37 @@ func getVolumeAndAttachment(volumeId string, attachmentId string) (*model.Volume
 }
 
 // mountDeviceAndUpdateAttachment Mount device and then update attachment
-func mountDeviceAndUpdateAttachment(device string, mountpoint string, key string, mountFlags []string, needUpdateAtc bool, attachment *model.VolumeAttachmentSpec) error {
+func mountDeviceAndUpdateAttachment(device string, mountpoint string, key string, mountFlags []string, needUpdateAtc bool, attachment *model.VolumeAttachmentSpec, block *csi.VolumeCapability_BlockVolume) error {
 	var err error
 
-	if len(mountFlags) > 0 {
-		_, err = exec.Command("mount", "-o", strings.Join(mountFlags, ","), device, mountpoint).CombinedOutput()
+	if nil == block {
+		if len(mountFlags) > 0 {
+			_, err = exec.Command("mount", "-o", strings.Join(mountFlags, ","), device, mountpoint).CombinedOutput()
+		} else {
+			_, err = exec.Command("mount", device, mountpoint).CombinedOutput()
+		}
+
+		if nil != err {
+			return status.Error(codes.Aborted, fmt.Sprintf("failed to mount: %v", err.Error()))
+		}
 	} else {
-		_, err = exec.Command("mount", device, mountpoint).CombinedOutput()
-	}
+		_, err = os.Lstat(mountpoint)
 
-	if nil != err {
-		return status.Error(codes.Aborted, fmt.Sprintf("failed to mount: %v", err.Error()))
+		if err != nil && os.IsNotExist(err) {
+			glog.V(5).Infof("Mountpoint=%v does not exist", mountpoint)
+		} else {
+			glog.Errorf("mountpoint=%v already exists", mountpoint)
+			_, err := exec.Command("rm", "-rf", mountpoint).CombinedOutput()
```

Comment:
If mount point already exists, is it safe to remove it here?  Is this to handle idempotency? In any case, please add comments to explain why this is needed and why this is safe to do.

Reply:
The mountpoint deleted here is a folder or a soft connection. From the test results, this is fine.